src/components/LDRouteGuard.vue

```diff
@@ -8,7 +8,7 @@ import ldRedirectMixin from '../mixins/ldRedirect';
 export default {
   mixins: [ldRedirectMixin()],
   props: {
-    component: { type: [Object, Promise], required: true },
+    component: { type: [Function, Object, Promise], required: true },
```

Comment:
Guess this is simpler than making folks promise-ify their functions.

Reply:
A dynamic import assigned to a const is a promise. But yeah, simpler ^_^.